### PR TITLE
CXP-1366: Update the product details page

### PR DESCRIPTION
**Pull request by eoliveauakeneo**
Update the product details page to display mapped products

image

### CODE REVIEW

**Comment by RiffVentura** [comment]
Since it is the last PR of the batch, it should remove previous CatalogID storage and associated code

**Review by RiffVentura — APPROVED** [review]

File: src/PimApi/PimCatalogApiClient.php [hunk 1]
```diff
@@ -219,4 +227,30 @@ public function getMappedProducts(
...
+        $this->throwOnErrorCode(200, $response->getStatusCode(), "Couldn't get mapped product");
+
+        $response = $response->toArray();
+
+        if (isset($response['message']) || isset($response['error'])) {
+            throw new CatalogDisabledException();
+        }
+
+        return $response;
+    }
```
Comment (hogren) [inline src/PimApi/PimCatalogApiClient.php:255]: I think that you can factorize the getCatalogProduct method. In my opinion, there is a quiet big duplicate code section here. What do you think @akeneo/squad-octopus ?

**Review by hogren — APPROVED** [review]

File: src/PimApi/PimCatalogApiClient.php (hunk 1 above)
Comment (eoliveauakeneo) [reply src/PimApi/PimCatalogApiClient.php:255]: We could factorize the majority of this page but I think we want the code as straightforward as possible. What is your pov @RiffVentura @masacc  ?

Comment (RiffVentura) [reply src/PimApi/PimCatalogApiClient.php:255]: What kind of refactoring do you have in mind?
Keep in mind that this whole class is temporary and will be scrapped as soon as [api-php-client](https://github.com/akeneo/api-php-client) supports all of our new catalog-related endpoints

Comment (masacc) [reply src/PimApi/PimCatalogApiClient.php:255]: I disagree to factorize this for all the reasons Etienne and Rifat evoked + in a CQRS way of thinking, it's not such a bad thing to have duplicate code if the contexts are different which is the case right here.a

Comment (tseho) [reply src/PimApi/PimCatalogApiClient.php:255]: Agree with the 3 others, DRY & code mutualization is only valid when **you are sure** that both pieces of code will always evolve the same way.

Here, they both answer to different needs and there is a high probably that `getMappedProduct` will evolve in a different direction, if there more errors to handle for example.

Comment (tseho) [reply src/PimApi/PimCatalogApiClient.php:255]: For the fun fact, mutualization of code is a trap we have fallen in for the PIM in the past.
This is why some legacy parts of the PIM are hard to work with.

We have proved in our more recent features that avoiding DRY principles, even if we have to slightly copy/paste some code is better in the long term.
I really like the current way of working of the octopus where we postpone mutualization for later to have a better view of how it should be organized.

**Pull request merged**